ros2plugin/ros2plugin/api/__init__.py

```diff
@@ -0,0 +1,67 @@
+# Copyright 2019 Canonical Ltd.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from ament_index_python.resources import get_resource
+from ament_index_python.resources import get_resource_types
+from ament_index_python.resources import get_resources
+from ament_index_python.resources import has_resource
+
+PLUGIN_RESOURCE_TYPE = '__pluginlib__plugin'
+
+
+def is_plugin_ressource_type(resource_type):
+    """Return True if resource_type has the plugin extension."""
+    assert resource_type, 'The resource type must not be empty'
+    return PLUGIN_RESOURCE_TYPE in resource_type
+
+
+def get_registered_plugin_ressource_list():
+    """Get all plugin ressources registered in the ament index."""
+    return filter(is_plugin_ressource_type, get_resource_types())
+
+
+def get_package_names_with_plugin_ressource_types():
+    """Get the names of all packages that register a plugin ressource in the ament index."""
```

Comment:
It would be good to document the return value.

Same above.

Reply:
Done.